rfcs/2020-04-04-2221-kubernetes-integration.md

```diff
@@ -0,0 +1,1625 @@
+and/or Helm Charts. For now, we should probably leave them as is, maintain them,
+but hold the adoption as integration tests.
+
+### Other data gathering
+
+> This section is on gathering data other than container logs.
+
+While our main focus for the integration is collecting log data from the `Pod`s,
+there are other possibilities to gain observability in the Kubernetes
+environment.
+
+#### Exposing Kubernetes [`Event`s][k8s_api_event] as Vector events
+
+It is possible to subscribe to Kubernetes [`Event`s][k8s_api_event], similarly
+to how this command works:
+
+```shell
+kubectl get events --all-namespaces --watch
+```
+
+Implementing this in Vector would allow capturing the Kubernetes
+[`Event`s][k8s_api_event] and processing them as Vector events.
+
+This feature might be very useful for anyone that wants to see what's going on
+in their cluster.
+
+Note that this feature would require deploying Vector in a differently: instead
+of running Vector on every node, here we need only once Vector instance running
+per cluster. If run on every node, it'd be unnecessarily capturing each event
+multiple times.
+
+So, to implement this, we'd need to add a special source that captures events
+from Kubernetes API, and provide a new workload configuration based on
+[`Deployment`][k8s_api_deployment].
+
+See also a section on collecting
+[Kubernetes audit logs][anchor_kubernetes_audit_logs].
+
+#### Discover and gather Prometheus metrics for Kubernetes API resources
+
+Prometheus already has a built-in
+[Kubernetes Service Discovery][prometheus_kubernetes_sd_config] support, so one
+could just deploy a Prometheus server, make it discover and gather the metrics,
+and the configure Vector to read metrics from it.
+
+However, to pursue our goal of making Vector the only agent one would need to
+deploy - we can consider reimplementing what prometheus
+[does][prometheus_kubernetes_sd_config] in Vector code, eliminate the need for
+the intermediary.
+
+We don't aim to implement this in the initial Kubernetes integration.
+
+#### Gather data from the host OS
+
+This is very useful for Kubernetes Cluster Operators willing to deploy Vector
+for the purposes of gaining observability on what's going on with their cluster
+nodes.
+
+Example use cases are:
+
+- reading `kubelet`/`docker` logs from `journald`;
+- capturing `kubelet`/`docker` prometheus metrics;
+- gathering system metrics from the node, things like `iostat -x`, `df -h`,
+  `uptime`, `free`, etc;
+- gathering system logs, like `sshd`, `dmesg` and etc.
+
+There are countless use cases here, and good news Vector already well fit to
+perform those kinds of tasks! Even without any Kubernetes integration
+whatsoever, it's possible to just deploy Vector as a
+[`DaemonSet`][k8s_api_daemon_set], expose the system data to it via
+[`hostPath` volume][k8s_api_host_path_volume_source] mounts and/or enabling
+`hostNetwork` at the [`PodSpec`][k8s_api_pod_spec].
```

Comment:
Please make sure this is in the plan of attack so that we ensure we include it in our docs. Is this enabled by default or will users have to configure this explicitly?